tests/test_mixing.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2023 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+from shutil import rmtree
+from typing import List
+
+from streaming import MDSWriter, Stream, StreamingDataset
+
+
+def walk(dataset: StreamingDataset) -> List[int]:
+    return [sample['value'] for sample in dataset]
+
+
+def float_eq(a: float, b: float) -> bool:
+    return abs(a - b) < 1e-6
+
+
+def test_mixing():
+    root = '/tmp/foo'
```

Comment:
you can use `local_remote_dir: Tuple[str, str]` which creates a tmp directory for you and also deletes automatically when test finishes. Please take a look at `test_upload.py` file for example.

Reply:
fixed